compaction.go

```diff
@@ -2509,13 +2516,34 @@ func (d *DB) runCopyCompaction(
 		}
 		pendingOutputs = append(pendingOutputs, newMeta)
 
-		if err := objstorage.Copy(ctx, src, w, 0, uint64(src.Size())); err != nil {
-			w.Abort()
-			return ve, pendingOutputs, err
-		}
-		if err := w.Finish(); err != nil {
-			return ve, pendingOutputs, err
+		if d.opts.Experimental.DownloadViaSpaCopy {
```

Comment:
I suspect a sizable number of externalSSTs will exactly match the physical sst. Could we add a fast path where we do an old style byte for byte copy if there's an exact match? This could be a follow up PR. I also wonder how much overhead this block copy approach takes-- if it's small, this extra fast path may not be worth it.

Reply:
I like this idea

Reply:
> I like this idea

Can you expand more on why?

I certainly don't think we should add it now, since it is strictly an optimization that we should let benchmarking tell us to add.

Unless we are only using some tiny number of blocks, the cost of the block copy is going to dominate the cost of building and writing the index. If we're copying a tiny number of blocks from a huge file, this block-copy+build-index is extremely worthwhile. If we're copying all the blocks from a large file, the added overhead of building an index vs copying it is negligible. If we're copying all the blocks from a tiny file, then sure, the overhead of building a new index _relative to the copy_ is something, but in the big picture of an online restore is inconsequential.

Reply:
No strong opinion when we should add it. I like it because it's very simple to implement (e.g. if the slice of blocks we need to copy is > 90% of the object, just copy everything) - much less work than it would take to benchmark things. I don't have data to go on so I don't know that the building of the index blocks is negligible. In aggregate, it might add a non-trivial CPU component to the process.